services/storage/src/simcore_service_storage/s3_client.py

```diff
@@ -237,17 +242,22 @@ async def complete_multipart_upload(
         file_id: SimcoreS3FileID,
         upload_id: UploadID,
         uploaded_parts: list[UploadedPart],
+        sha256_checksum: SHA256Str | None,
     ) -> ETag:
-        response = await self.client.complete_multipart_upload(
-            Bucket=bucket,
-            Key=file_id,
-            UploadId=upload_id,
-            MultipartUpload={
+        inputs: dict[str, Any] = {
+            "Bucket": bucket,
+            "Key": file_id,
+            "UploadId": upload_id,
+            "MultipartUpload": {
                 "Parts": [
                     {"ETag": part.e_tag, "PartNumber": part.number}
                     for part in uploaded_parts
                 ]
             },
+            "ChecksumSHA256": sha256_checksum,
+        }
+        response = await self.client.complete_multipart_upload(
+            **{k: v for k, v in inputs.items() if v is not None}
```

Comment:
Q: instead of adding and removing items, would it not be much cleaner to add them if needed? I see you use this pattern elsewhere. It will also avoid a few extra ifs.

Reply:
Hmm, ok fair point. I changed this so that the checksum is only added to the kwargs dict if is not None.